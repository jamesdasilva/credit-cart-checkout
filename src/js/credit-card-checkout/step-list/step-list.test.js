import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import React from 'react'

import StepList from './step-list';

describe('<StepList /> spec', () => {
  test('Snapshot test', () => {
    const objectMock = { 
      steps: [
        "Carrinho",
        "Pagamento",
        "Confirmação"
      ],
      stepSelected: "Carrinho"
    }
    const { getByTestId } = render(
      <StepList
        steps={objectMock.steps}
        stepSelected={objectMock.stepSelected} />
      )

    const stepList = getByTestId('step-list')
    expect(stepList).toMatchSnapshot()
  })

  test('Deve exibir os nomes das steps nos itens da lista', () => {
    const objectMock = {
      steps: [
        "Carrinho",
        "Pagamento",
        "Confirmação"
      ],
      stepSelected: "Carrinho"
    }
    const { getByTestId } = render(
      <StepList
        steps={objectMock.steps}
        stepSelected={objectMock.stepSelected} />
      )

    const itemListCount = getByTestId('step-list').childNodes
    
    const itemTexts = [...itemListCount].map(item => {
      return item.textContent.match(/[ A-Za-záàâãéèêíïóôõöúçñÁÀÂÃÉÈÍÏÓÔÕÖÚÇÑ]+/)[0]
    })

    expect(itemTexts).toEqual(objectMock.steps)
  })

  test('Deve não exibir check-icon nenhum quando stepSelected == steps[0]', () => {
    const objectMock = {
      steps: [
        "Carrinho",
        "Pagamento",
        "Confirmação"
      ],
      stepSelected: "Carrinho"
    }
    const { queryAllByTestId, queryByText } = render(
      <StepList
        steps={objectMock.steps}
        stepSelected={objectMock.stepSelected} />
      )

    const itemIcon = queryAllByTestId('check-icon')
    expect(itemIcon.length).toBeFalsy();

    const item1 = queryByText(/[1]+/)
    const item2 = queryByText(/[2]+/)
    const item3 = queryByText(/[3]+/)
    expect(item1).toBeTruthy();
    expect(item2).toBeTruthy();
    expect(item3).toBeTruthy();

  })

  test('Deve exibir check-icon no primeiro item quando stepSelected == steps[1]', () => {
    const objectMock = {
      steps: [
        "Carrinho",
        "Pagamento",
        "Confirmação"
      ],
      stepSelected: "Pagamento"
    }
    const { queryByTestId, queryByText } = render(
      <StepList
        steps={objectMock.steps}
        stepSelected={objectMock.stepSelected} />
      )

    const itemIcon = queryByTestId('check-icon')
    expect(itemIcon).toBeTruthy();

    const item1 = queryByText(/[1]+/)
    const item2 = queryByText(/[2]+/)
    const item3 = queryByText(/[3]+/)
    expect(item1).toBeFalsy();
    expect(item2).toBeTruthy();
    expect(item3).toBeTruthy();

  })

  test('Deve exibir check-icon nos dois primeiros itens quando stepSelected == steps[2]', () => {
    const objectMock = {
      steps: [
        "Carrinho",
        "Pagamento",
        "Confirmação"
      ],
      stepSelected: "Confirmação"
    }
    const { queryAllByTestId, queryByText } = render(
      <StepList
        steps={objectMock.steps}
        stepSelected={objectMock.stepSelected} />
      )

    const itemIcon = queryAllByTestId('check-icon')
    expect(itemIcon.length).toBe(2);

    const item1 = queryByText(/[1]+/)
    const item2 = queryByText(/[2]+/)
    const item3 = queryByText(/[3]+/)
    expect(item1).toBeFalsy();
    expect(item2).toBeFalsy();
    expect(item3).toBeTruthy();

  })
})
