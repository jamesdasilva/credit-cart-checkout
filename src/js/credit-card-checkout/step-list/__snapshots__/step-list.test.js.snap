// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepList /> spec Snapshot test 1`] = `
<div
  class="step-list"
  data-testid="step-list"
>
  <div
    class="step-list__item"
  >
    <div
      class="step-list__number"
    >
      1
    </div>
    Carrinho
    <div
      class="step-list__separator"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-angle-right fa-w-8 fa-lg "
        data-icon="angle-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
  <div
    class="step-list__item"
  >
    <div
      class="step-list__number"
    >
      2
    </div>
    Pagamento
    <div
      class="step-list__separator"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-angle-right fa-w-8 fa-lg "
        data-icon="angle-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
  <div
    class="step-list__item"
  >
    <div
      class="step-list__number"
    >
      3
    </div>
    Confirmação
  </div>
</div>
`;
